app/main.py

```diff
@@ -1,3 +1,66 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float | Distance) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(
+                km=self.km + other.km
+            )
+
+        return Distance(
+            km=self.km + other
+        )
+
+    def __iadd__(self, other: int | float | Distance) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        return self
+
+    def __mul__(self, other: int | float | Distance) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(
+                km=self.km * other
+            )
+
+    def __truediv__(self, other: int | float | Distance) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(
+                km=round(self.km / other, 2)
+            )
+
+    def __lt__(self, other: int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km < other.km
+        return self.km < other
+
+    def __gt__(self, other: int | float) -> bool:
```

Comment:
The type hint for the `other` parameter in the comparison methods should include `Distance` to support comparisons with `Distance` objects.